src/mount.ts

```diff
@@ -73,11 +71,24 @@ export type ObjectEmitsOptions = Record<
 >
 export type EmitsOptions = ObjectEmitsOptions | string[]
 
-// Class component
-export function mount(
-  originalComponent: VueConstructor,
+// Class component - no props
+export function mount<V>(
+  originalComponent: {
+    new (...args: any[]): V
+    registerHooks(keys: string[]): void
+  },
   options?: MountingOptions<any>
-): VueWrapper<ComponentPublicInstance<any>>
+): VueWrapper<ComponentPublicInstance<V>>
+
+// Class component - props
+export function mount<V, P>(
+  originalComponent: {
+    new (...args: any[]): V
+    props(Props: P): any
+    registerHooks(keys: string[]): void
+  },
+  options?: MountingOptions<P>
+): VueWrapper<ComponentPublicInstance<V>>
```

Comment:
```suggestion
export function mount<V extends ComponentPublicInstance<P>, P>(
  originalComponent: {
    new (...args: any[]): V
  },
  options?: MountingOptions<P>
): VueWrapper<V>
```

As class component implements component public instance, just using the interface should be enough.

Reply:
That's a bit too lenient, because the `defineComponent` will also return a valid component public instance to be used on JSX

Reply:
Yep, agree, having `props` intellisense is pretty cool. I don't think it's possible for `data` in class component though.